msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/BiGaussian.java

```diff
@@ -0,0 +1,198 @@
+		
+		listOfIntensities = new ArrayList<Double>();
+		
+		for(int i=0;i<listOfTripletsInSlice.size();i++){
+			double intensity = listOfTripletsInSlice.get(i)!=null?listOfTripletsInSlice.get(i).intensity:0;
+			listOfIntensities.add(intensity);
+		}
+		
+		
+		
+		Comparator<Double> compareIntensities = new Comparator<Double>() {
+			
+			@Override
+			public int compare(Double o1, Double o2) {
+				Double intensity1 = o1;
+				Double intensity2 = o2;
+				return intensity1.compareTo(intensity2);
+			}
+		};
+		
+		Collections.sort(listOfIntensities, compareIntensities);
+		
+		//This is max height for BiGaussian fit. It's in terms of intensities.
+		double maxHeight = listOfIntensities.get(listOfIntensities.size()-1);
+		
+		//Below logic is for finding BiGaussian parameters.		
+		int mu = getScanNumber(maxHeight);
+		int roundedmz = (int) Math.round(mz * 10000); 
+		double halfHeight = (double) maxHeight/2; 
+		biGaussianParams = new double[4];
+		
+		biGaussianParams[0] = maxHeight;
+		biGaussianParams[1] = mu;
+	
+		double  interpolationLeftSideY1 = 0;
+		double interpolationLeftSideY2 = 0;
+		
+		for(int i=mu-1;i>=leftBound;i--){
+			
+			SliceSparseMatrix.Triplet triplet1 = ((SliceSparseMatrix.Triplet)horizontalSlice.get(i,roundedmz));
+			
+			if(triplet1!=null){
+				interpolationLeftSideY1 = triplet1.intensity;
+			}
+			
+			if(interpolationLeftSideY1 <halfHeight && triplet1!=null){
+				SliceSparseMatrix.Triplet triplet2 =  ((SliceSparseMatrix.Triplet)horizontalSlice.get(i+1,roundedmz));
+				if(triplet2!=null){
+					interpolationLeftSideY2 = triplet2.intensity;
+					break;
+				}
+			}
+		}
+		
+		double interpolationLeftSideX = ((halfHeight - interpolationLeftSideY1)*(getScanNumber(interpolationLeftSideY2)-getScanNumber(interpolationLeftSideY1))
+				/(interpolationLeftSideY2 - interpolationLeftSideY1))+getScanNumber(interpolationLeftSideY1);
+		
+		//This is signa left for BiGaussian. 
+		biGaussianParams[2] = (mu - interpolationLeftSideX)/Math.sqrt(2*Math.log(2)); 
+		
+		double  interpolationRightSideY1 = 0;
+		double interpolationRightSideY2 = 0;
+		
+		for(int i=mu+1;i<=rightBound;i++){
+			
+			SliceSparseMatrix.Triplet triplet1 = ((SliceSparseMatrix.Triplet)horizontalSlice.get(i,roundedmz));
+			
+			if(triplet1!=null){
+				interpolationRightSideY1 = triplet1.intensity;
+			}
+			
+			if(interpolationRightSideY1 <halfHeight && triplet1!=null){
+				SliceSparseMatrix.Triplet triplet2 =  ((SliceSparseMatrix.Triplet)horizontalSlice.get(i-1,roundedmz));
+				if(triplet2!=null){
+					interpolationRightSideY2 = triplet2.intensity;
+					break;
+				}
+			}
+				
+		}
+		
+		double interpolationRightSideX = ((halfHeight - interpolationRightSideY1)*(getScanNumber(interpolationRightSideY2)-getScanNumber(interpolationRightSideY1))
+				/(interpolationLeftSideY2 - interpolationRightSideY1))+getScanNumber(interpolationRightSideY1);
+		
```

Comment:
The two ```for``` loops seem essentially identical except for the iteration bounds (i.e. ```mu-1 .. leftBound``` vs. ```mu+1 .. rightBound```). Suggest refactoring into a single function that takes the bounds you iterate over and returns the ```interpoloated*SideX```.